app/models/score.rb

```diff
@@ -1,3 +1,56 @@
 class Score < ActiveRecord::Base
   belongs_to :subscriber
+
+  def self.count_intervals(interval_length)
+    first_created = all.last.created_at
+    last_created = all.first.created_at
+    minutes_apart = (
+      (last_created.minus_with_coercion(first_created)
+      ).floor / 60)
+    (minutes_apart / interval_length).floor
+  end
+
+  # Returns the average score of records within a given interval in minutes
+  #
+  def self.interval_average(int_begin, int_end)
+    scores = all.where(created_at: int_begin..int_end)
+    return 0 unless scores.present?
+    scores.average(:score).floor
+  end
+
+  # Returns a hash. The value is the average score for the interval length
+  # given in minutes starting from the time indicated by the key
+  #
+  def self.interval_averages(interval_length)
+    result = {}
+    int_begin = all.last.created_at
+    (0..all.count_intervals(interval_length)).each do
+      int_end = int_begin + interval_length.minutes
+      average = all.interval_average(int_begin, int_end)
+      result.merge!(int_begin.strftime('%I:%M%p') => average)
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.